lib/logstash/filters/elasticsearch.rb

```diff
@@ -208,7 +218,7 @@ def filter(event)
           set = resultsHits.map do |doc|
             extract_value(doc["_source"], old_key_path)
           end
-          event.set(new_key, set.count > 1 ? set : set.first)
+          set_to_event_target(event, new_key, set)
```

Comment:
review discussion:
- similar situation with `docinfo_fields` that we can introduce `docinfo_target` but this will be ECSified and default would be `v1: '[@metadata][filter][elasticsearch]'`
- where to place L211 "[@metadata][total_hits]" is also another question

Reply:
The difference between `fields` and `docinfo_fields` is the _source_ of the field -- `fields` sources the _indexed_ fields, while `docinfo_fields` sources the values as-present in the document's `_source`.

But the behavior of the destination is the same, so I think that the two can (and _should_) safely share a `target`, which would get prepended to both.

If a user wants to store them in separate places, they can continue to use the existing pattern of specifying fully-qualified `fields` and `docinfo_fields` mappings separately with no `target`.